tests/test_trainer.py

```diff
@@ -411,5 +413,85 @@ class CurrentTestModel(
     trainer.test()
 
 
+def _init_steps_model():
+    """private method for initializing a model with 5% train epochs"""
+    tutils.reset_seed()
+    model, _ = tutils.get_model()
+
+    # define train epoch to 5% of data
+    train_percent = 0.05
+    # get number of samples in 1 epoch
+    num_train_samples = math.floor(len(model.train_dataloader()) * train_percent)
+
+    trainer_options = dict(
+        train_percent_check=train_percent,
+    )
+    return model, trainer_options, num_train_samples
+
+
+def test_trainer_max_steps_and_epochs(tmpdir):
+    """Verify model trains according to speficied max steps"""
+    model, trainer_options, num_train_samples = _init_steps_model()
+
+    # define less train steps than epochs
+    trainer_options['max_epochs'] = 5
+    trainer_options['max_steps'] = num_train_samples + 10
+
+    # fit model
+    trainer = Trainer(**trainer_options)
+    result = trainer.fit(model)
+
+    # check training stopped at max_steps
+    assert result == 1
+    assert trainer.global_step == trainer_options['max_steps'], "Model did not stop at max_steps"
+
+    # define less train epochs than steps
+    trainer_options['max_epochs'] = 2
+    trainer_options['max_steps'] = trainer_options['max_epochs'] * 2 * num_train_samples
+
+    # fit model
+    trainer = Trainer(**trainer_options)
+    result = trainer.fit(model)
+    assert result == 1
+
+    # check training stopped at max_epochs
+    assert trainer.global_step == num_train_samples * trainer_options['max_epochs'] and \
+        trainer.current_epoch == trainer_options['max_epochs'] - 1, "Model did not stop at max_epochs"
+
+
+def test_trainer_min_steps_and_epochs(tmpdir):
+    """Verify model trains according to speficied min steps"""
+    model, trainer_options, num_train_samples = _init_steps_model()
+
+    # define callback for stopping the model and default epochs
+    trainer_options['early_stop_callback'] = EarlyStopping(monitor='val_loss', min_delta=1.0)
```

Comment:
personally I would squash the parameters so it is clear that it is still just one object
```python
trainer_options = dict(
    early_stop_callback=EarlyStopping(monitor='val_loss', min_delta=1.0),
    val_check_interval=20,
    min_epochs=1,
    max_epochs=10,
    ...
)
```

Reply:
Done.